File: 51-60/Q57.cpp
//Q57.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& a, vector<int>& n) {
        vector<vector<int>> res;
        int i=0, m=a.size();
        while(i<m && a[i][1]<n[0]) res.push_back(a[i++]);
        while(i<m && a[i][0]<=n[1]){
            n[0]=min(n[0], a[i][0]);
            n[1]=max(n[1], a[i][1]);
            i++;
        }
        res.push_back(n);
        while(i<m) res.push_back(a[i++]);
        return res;
    }
};

File: 51-60/Q57.py
#Q57.py
class Solution:
    def insert(self, a, n):
        res=[]; i=0; m=len(a)
        while i<m and a[i][1]<n[0]:
            res.append(a[i]); i+=1
        while i<m and a[i][0]<=n[1]:
            n[0]=min(n[0], a[i][0])
            n[1]=max(n[1], a[i][1])
            i+=1
        res.append(n)
        while i<m:
            res.append(a[i]); i+=1
        return res
